pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/LanguageRuleMapping.java

```diff
@@ -0,0 +1,87 @@
+package sfdc.sfdx.scanner.pmd;
+
+import java.util.*;
+
+/**
+ * Maintains mapping between languages and their XML paths
+ */
+public class LanguageRuleMapping {
+  private final static String CATEGORY = "category";
+  private final static String RULESETS = "rulesets";
+
+  private static LanguageRuleMapping INSTANCE = new LanguageRuleMapping();
+
+  private Map<String, Set<String>> categoryPathsByLanguage;
+  private Map<String, Set<String>> rulesetPathsByLanguage;
+
+  private LanguageRuleMapping() {
+    categoryPathsByLanguage = new HashMap<>();
+    rulesetPathsByLanguage = new HashMap<>();
+  }
+
+  public static LanguageRuleMapping getInstance() {
+    return INSTANCE;
+  }
+
+  public void addPathsForLanguage(List<String> paths, String language) {
+    paths.forEach(path -> addPathForLanguage(path, language));
+  }
+
+  public Map<String, Set<String>> getCategoryPaths() {
+    return this.categoryPathsByLanguage;
+  }
+
+  public Map<String, Set<String>> getRulesetPaths() {
+    return this.rulesetPathsByLanguage;
+  }
+
+  void addPathForLanguage(String path, String language) {
+    if (! nullEmptyOrWhitespace(path)) {
+      if (path.contains(RULESETS)) {
+        addPathForLanguageToRulesets(path, language);
+      } else if (path.contains(CATEGORY)) {
+        addPathForLanguageToCategory(path, language);
+      } else {
+        // TODO: temporarily default to Category. But this won't work in all cases.
```

Comment:
I need to remove the last `else`, since we decided we won't consider an XML that hasn't been placed under `rulesets` or `category` directory structure.